rebac-admin-backend/v1/core.go

```diff
@@ -9,14 +9,24 @@ import (
 	"net/http"
 	"strings"
 
+	"github.com/canonical/identity-platform-admin-ui/rebac-admin-backend/v1/interfaces"
 	"github.com/canonical/identity-platform-admin-ui/rebac-admin-backend/v1/resources"
 )
 
 // ReBACAdminBackendParams contains references to user-defined implementation
 // of required abstractions, called "backend"s.
 type ReBACAdminBackendParams struct {
-	GroupsService       GroupsServiceBackend
-	GroupsAuthorization GroupsAuthorizationBackend
+	IdentitiesService       interfaces.IdentitiesService
+	IdentitiesAuthorization interfaces.IdentitiesAuthorization
+	IdentitiesErrorMapper   ErrorResponseMapper
+
+	RolesService       interfaces.RolesService
+	RolesAuthorization interfaces.RolesAuthorization
+	RolesErrorMapper   ErrorResponseMapper
+
+	GroupsService       interfaces.GroupsService
+	GroupsAuthorization interfaces.GroupsAuthorization
+	GroupsErrorMapper   ErrorResponseMapper
```

Comment:
suggestion: can we shorten the field names like in the `handler` struct?
example: `IdentitiesService` -> `Identities`

Reply:
I'm okay with that. Do you want to make the change here, or in another polish-like PR?

Reply:
I would do it here but feel free to do it later
I'll link this comment in #205 also just to keep track

Reply:
Done.